src/value/use-viewport-scroll.ts

```diff
@@ -96,10 +132,46 @@ const viewportMotionValues: ScrollMotionValues = {
  *
  * @public
  */
-export function useViewportScroll() {
-    if (!hasEventListener) {
-        addScrollListener()
+export function useViewportScroll(ref: undefined | RefObject<Element>) {
```

Comment:
In this case typescript will require passing either undefined or refObject. If you want optional value please use `?`

``` typescript 
export function useViewportScroll(ref?: RefObject<Element>)
```

otherwise this would throw an error:
``` typescript
const scroll = useViewportScroll() 
```